Automattic/wp-calypso
# Full Site Editing: Add a Template Part block

#### Changes proposed in this Pull Request

Note: this is based off #32529, so it will need to be rebased when that's merged.

* Add a Template Part block.
It roughly works like the Content Slot block but, whereas that's a sort of placeholder for the template to be filled with a post or page, the Template Part is used to render a layout element into the template.

The screenshot is basically the same (of course with a different title and label):

![Apr-23-2019 18-47-49](https://user-images.githubusercontent.com/2070010/56604673-745d1f00-65fa-11e9-9374-d65f4e5adecd.gif)

The main visible difference is that this renders the selected post's content on the front-end.

#### Testing instructions



* Follow the [README](https://github.com/Automattic/wp-calypso/blob/master/apps/full-site-editing/README.md) to build the plugin. (If the script fails, please run `npm distclean` and `npm ci` first).
* In the block editor, try adding the Template Part block.
* Type a post name (one with content 🙂) and select it.
* Make sure its content shows up in the editor.
* Save and preview (or view) the post.
* Make sure the preview contains the selected post's content.

Fixes #32493
